src/dex/cables/cables.ts

```diff
@@ -0,0 +1,1006 @@
+
+    return {
+      exchangeData,
+      needWrapNative: this.needWrapNative,
+      dexFuncHasRecipient: false,
+      targetExchange: this.mainnetRFQAddress,
+      returnAmountPos: undefined,
+      insertFromAmountPos: filledAmountPos,
+    };
+  }
+
+  getAdapterParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: CablesData,
+    side: SwapSide,
+  ): AdapterExchangeParam {
+    const { quoteData } = data;
+
+    assert(
+      quoteData !== undefined,
+      `${this.dexKey}-${this.network}: quoteData undefined`,
+    );
+
+    const params = [
+      {
+        nonceAndMeta: quoteData.nonceAndMeta,
+        expiry: quoteData.expiry,
+        makerAsset: quoteData.makerAsset,
+        takerAsset: quoteData.takerAsset,
+        maker: quoteData.maker,
+        taker: quoteData.taker,
+        makerAmount: quoteData.makerAmount,
+        takerAmount: quoteData.takerAmount,
+      },
+      quoteData.signature,
+    ];
+
+    const payload = this.abiCoder.encodeParameter(
+      {
+        ParentStruct: {
+          order: {
+            nonceAndMeta: 'uint256',
+            expiry: 'uint128',
+            makerAsset: 'address',
+            takerAsset: 'address',
+            maker: 'address',
+            taker: 'address',
+            makerAmount: 'uint256',
+            takerAmount: 'uint256',
+          },
+          signature: 'bytes',
+        },
+      },
+      {
+        order: params[0],
+        signature: params[1],
+      },
+    );
+
+    return {
+      targetExchange: this.mainnetRFQAddress,
+      payload,
+      networkFee: '0',
+    };
+  }
+
+  normalizeToken(token: Token): Token {
+    return {
+      ...token,
+      address: this.normalizeTokenAddress(token.address),
+    };
+  }
+
+  normalizeTokenAddress(address: Address): Address {
+    return address.toLowerCase();
+  }
+
+  getTokenFromAddress(address: Address): Token {
+    return this.tokensMap[this.normalizeAddress(address)];
+  }
+
+  getPoolIdentifier(srcAddress: Address, destAddress: Address, mm?: string) {
+    return `${this.dexKey}_${srcAddress}_${destAddress}`.toLowerCase();
+    return `${this.dexKey}_${srcAddress}_${destAddress}_${mm}`.toLowerCase();
+  }
+
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    await this.setTokensMap();
+
+    if (!srcToken || !destToken) {
+      return [];
+    }
+
+    if (srcToken.address.toLowerCase() === destToken.address.toLowerCase()) {
+      return [];
+    }
+
+    const pairData = await this.getPairData(srcToken, destToken);
+
+    if (!pairData) {
+      return [];
+    }
+
+    const tokensAddr = (await this.getCachedTokensAddr()) || {};
+
+    return [
+      this.getPoolIdentifier(
+        tokensAddr[pairData.base.toLowerCase()],
+        tokensAddr[pairData.quote.toLowerCase()],
+      ),
+    ];
```

Comment:
In one `getPoolIdentifiers` we do `4` Redis calls. I think it's too much and should be optimized (mostly because there's no sense in making same Redis calls)

Reply:
fixed